packages/react-day-picker/src/contexts/Focus/getInitialTargetFocusedDay.tsx

```diff
@@ -0,0 +1,38 @@
+import { addDays, endOfMonth, startOfMonth } from 'date-fns';
+
+import { getModifierStatus } from 'hooks/useDayModifiers/utils/getModifierStatus';
+import { Modifiers } from 'types/Modifiers';
+
+export function getInitialTargetFocusedDay(
+  displayMonths: Date[],
+  modifiersContext: Modifiers
+) {
+  const firstDayInMonth = startOfMonth(displayMonths[0]);
+  const lastDayInMonth = endOfMonth(displayMonths[displayMonths.length - 1]);
+
+  let firstEnabledDay = undefined;
+  let today = undefined;
+  let date = firstDayInMonth;
+  while (date <= lastDayInMonth) {
+    const modifiers = getModifierStatus(date, modifiersContext);
+    if (modifiers.disabled || modifiers.hidden) {
+      date = addDays(date, 1);
+      continue;
+    }
+    if (modifiers.selected) {
+      return date;
+    }
+    if (modifiers.today && !today) {
+      today = date;
+    }
+    if (!firstEnabledDay) {
+      firstEnabledDay = date;
+    }
+    date = addDays(date, 1);
+  }
+  if (today) {
+    return today;
+  } else {
+    return firstEnabledDay;
+  }
```

Comment:
This is the code that implements the requirements, with the added complexity of handling disabled and hidden days, multiple months, and multiple selected days.

> if the input field is empty, or does not contain a valid date, 
> then the current date is focused in the calendar. Otherwise, 
> the focus is placed on the day in the calendar that matches 
> the value of the date input field.

https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html

Reply:
Love it ! good work!